examples/image-type-scala-pi.yaml

```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: spark.apache.org/v1alpha1
+kind: SparkApplication
+metadata:
+  name: pi-scala
+spec:
+  mainClass: "org.apache.spark.examples.SparkPi"
```

Comment:
`org.apache.spark.examples.SparkPi` is already used as a main example.

https://github.com/apache/spark-kubernetes-operator/blob/0f6011cccae003a0e9fb5b58ea1bf98ac78e54d8/examples/pi.yaml#L20

https://github.com/apache/spark-kubernetes-operator/blob/0f6011cccae003a0e9fb5b58ea1bf98ac78e54d8/examples/submit-pi-to-prod.sh#L36

Python example will be better, @jiangzho .

Reply:
Of the 4 Spark docker image type (scala, Python, R, all), we are now serving examples using the first 2 types. In other words, images with / without Python.

As the image with Python is used in all other examples, this example shows usage for the scala image and therefore it leverages `org.apache.spark.examples.SparkPi` . 

Or, would you like to refactor other examples to use the scala image while keeping only the python example using the Python image?

Reply:
Well, I disagree the idea of refactoring because the other images are intentionally use the shortest image name.